app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/FahrplanFragment.java

```diff
@@ -298,20 +305,45 @@ private void updateNavigationMenuSelection() {
         }
     }
 
-    private void addRoomColumns(HorizontalSnapScrollView scroller) {
-        LinearLayout root = (LinearLayout) scroller.getChildAt(0);
-        root.removeAllViews();
-        if (scroller.getColumnWidth() != 0) {
-            // update pre-calculated width with actual layout
-            columnWidth = scroller.getColumnWidth();
+    private void addRoomColumns(@NonNull HorizontalSnapScrollView horizontalScroller, @NonNull List<Lecture> lectures, boolean forceReload) {
+        int columnIndexLeft = horizontalScroller.getColumn();
+        int columnIndexRight = columnIndexLeft + maxRoomColumnsVisible;
+
+        // whenever possible, just update recycler views
+        if (!forceReload && !adapterByRoom.isEmpty()) {
+            for (int columnIndex = columnIndexLeft; columnIndex <= columnIndexRight; columnIndex++) {
+                int roomIndex = MyApp.roomList.get(columnIndex);
```

Comment:
:gift: I am currently working on to localize usage of the static `MyApp` fields to `FahrplanFragment#viewDay`. Could you please do me a favor and pass `MyApp.roomList` as a parameter to the `addRoomColumns` method to support my efforts? A second usage is down in line 331.

Reply:
The same would apply to roomCount. But I think having too many arguments for this method is not good, at this moment. Removing MyApp dependencies better suits in another PR.

Reply:
I find the method easier to understand when all input parameters are passed in via the method signature. Now one has to spot these external references to understand the routine. In tests one would have to manipulate the static fields in order to be able to test the method. - I am not happy with it but we can leave it as is for now.